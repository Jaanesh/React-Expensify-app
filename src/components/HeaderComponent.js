import React from 'react';
import {NavLink} from 'react-router-dom';
import {connect} from 'react-redux' ;
import { startLogout } from '../actions/auth';

export const Header=({startLogout})=>(
    <div>
        <h2>Expensify</h2>  
        <NavLink to='/dashboard' activeClassName="is-active" exact={true}>Dashboard</NavLink>
        <br/>
        <NavLink to='/create' activeClassName="is-active">create Expense</NavLink>
        <br/>
        <NavLink to='/edit' activeClassName="is-active">edit expense</NavLink> 
        <br/>
        <NavLink to='/help' activeClassName="is-active">help</NavLink>  
        <br/>
        <button onClick={startLogout}>Logout</button>    
    </div>   
);

const mapDispatchToProps=(dispatch)=>({
    startLogout:()=>dispatch(startLogout())
}  
)

export default connect(undefined,mapDispatchToProps)(Header);