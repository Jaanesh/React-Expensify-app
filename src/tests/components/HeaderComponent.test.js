import React from 'react';
import {shallow} from 'enzyme';
//import toJSON from 'enzyme-to-json';
//import ReactShallowRenderer from 'react-test-renderer/shallow';
import {Header}from '../../components/HeaderComponent.js';


test('should render header correctly',()=>{
        const wrapper=shallow(<Header startLogout={()=>{}} />);
        expect(wrapper).toMatchSnapshot();
        //expect(toJSON(wrapper)).toMatchSnapshot();

       // expect(wrapper.find('h2').text()).toBe('Expensify');
       /*const reactShallowRendererObject=new ReactShallowRenderer();
         reactShallowRendererObject.render(<Header/>);
         expect(reactShallowRendererObject.getRenderOutput()).toMatchSnapshot();
         console.log(reactShallowRendererObject.getRenderOutput());*/
});

test('should call start logout on button click',()=>{
      let startLogout=jest.fn();
      const wrapper=shallow(<Header startLogout={startLogout} />);
      wrapper.find('button').simulate('click');
      expect(startLogout).toHaveBeenCalled();
});
