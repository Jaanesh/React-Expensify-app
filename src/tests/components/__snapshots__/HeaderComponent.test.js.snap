// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render header correctly 1`] = `
<div>
  <h2>
    Expensify
  </h2>
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    exact={true}
    to="/dashboard"
  >
    Dashboard
  </NavLink>
  <br />
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    to="/create"
  >
    create Expense
  </NavLink>
  <br />
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    to="/edit"
  >
    edit expense
  </NavLink>
  <br />
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    to="/help"
  >
    help
  </NavLink>
  <br />
  <button
    onClick={[Function]}
  >
    Logout
  </button>
</div>
`;
